broker/providers/foreman.py

```diff
@@ -0,0 +1,427 @@
+"""Foreman provider implementation."""
+import time
+from uuid import uuid4
+
+import click
+from dynaconf import Validator
+from logzero import logger
+import requests
+
+from broker.exceptions import ProviderError
+from broker.helpers import Result
+from broker.providers import Provider
+from broker.settings import settings
+
+
+class ForemanAPI:
+    """Default runtime to query Foreman."""
+
+    headers = {
+        "Content-Type": "application/json",
+    }
+
+    def __init__(self, **kwargs):
+        self.foreman_username = settings.foreman.foreman_username
+        self.foreman_password = settings.foreman.foreman_password
+        self.url = settings.foreman.foreman_url
+        self.prefix = settings.foreman.name_prefix
+        self.verify = settings.foreman.verify
+        self.session = requests.session()
+
+    def interpret_response(self, response):
```

Comment:
Since this wouldn't be a "public" method, let's use a leading underscore.
```suggestion
    def _interpret_response(self, response):
```

Reply:
Done that